lib/cocoapods/installer/analyzer.rb

```diff
@@ -583,13 +593,19 @@ def generate_pod_targets(resolver_specs_by_target, target_inspections)
           end
           pod_targets.each do |target|
             all_specs = all_resolver_specs.group_by(&:name)
-            dependencies = dependencies_for_specs(target.non_test_specs.to_set, target.platform, all_specs.dup).group_by(&:root)
+            dependencies = dependencies_for_specs(target.library_specs.to_set, target.platform, all_specs.dup).group_by(&:root)
             target.dependent_targets = filter_dependencies(dependencies, pod_targets_by_name, target)
             target.test_dependent_targets_by_spec_name = target.test_specs.each_with_object({}) do |test_spec, hash|
               test_dependencies = dependencies_for_specs([test_spec], target.platform, all_specs).group_by(&:root)
               test_dependencies.delete_if { |k| dependencies.key? k }
               hash[test_spec.name] = filter_dependencies(test_dependencies, pod_targets_by_name, target)
             end
+
+            target.app_dependent_targets_by_spec_name = target.app_specs.each_with_object({}) do |app_spec, hash|
```

Comment:
would love to consolidate this into a single method since it does the same for test specs.